### PR TITLE
Smart Linking MVP

### PR DESCRIPTION
**Pull request by orangejenny**
## Summary
https://dimagi-dev.atlassian.net/browse/USH-1119

This is pretty minimal. It just sets up a URL route in HQ that redirects to an endpoint. It has no user-facing changes, since these links aren't yet being included anywhere. It will fail in a number of situations: user isn't logged in as a mobile worker, link is to a case that isn't in the user's casedb, etc.

Eventually these links may be used in reports, including external reports like in Tableau, and web apps. Full feature spec is [here](https://docs.google.com/document/d/1w7oRuH9skVhNB1vuhIa8XWbSBoUWNj3f_N6lnU_6lnc/edit#).

FYI @shubham1g5 

## Feature Flag
Enable session endpoints

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

None - this is all `FormplayerFrontEnd` web apps UI code that we don't have a test framework for and/or HQ view-level code.

### QA Plan

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-3278)

### Safety story
This is largely new functionality that isn't yet user-facing. It does touch the web apps UI, though the changes to existing code are small.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js [hunk 1]
```diff
@@ -405,8 +405,17 @@ hqDefine("cloudcare/js/formplayer/app", function () {
...
+        }
+        try {
+            var options = JSON.parse(url);
+            if (_.has(options, "endpointId")) {
+                return;
+            }
+        } catch (e) {
+            // do nothing
         }
+        window.history.back();
```
Comment (snopoke) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js:418]: this seems to invert the existing functionality (go back unless url includes 'single_app')

**Review by snopoke — APPROVED** [review]

**Comment by mkangia** [comment]
can we add some description about what these links would be? in PR or in the ticket.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js:418]: 🤦  thanks

e3a4938e379

**Comment by orangejenny** [comment]
@mkangia added spec link to PR description

**Comment by orangejenny** [comment]
Requested QA and updated safety info in PR description. QA will cover this PR and the sub PRs for claiming cases and redirecting to Login As.

**Pull request merged**